runtime/src/iree/hal/drivers/cuda/CMakeLists.txt

```diff
@@ -94,3 +104,19 @@ iree_cc_test(
   LABELS
     "driver=cuda"
 )
+
+if(IREE_HAL_DRIVER_CUDA_NCCL)
+iree_cc_test(
+  NAME
+    dynamic_symbols_test_for_nccl
+  SRCS
+    "dynamic_symbols_test_for_nccl.cc"
```

Comment:
you can use the IREE_HAL_DRIVER_CUDA_NCCL in dynamic_symbols.c to guard just that test in an #ifdef - that way we don't need an additional conditional test rule

Reply:
No more macros! I put the test in dynamic_symbols.c.